docs/user_docs/concepts/package-and-module.md

```diff
@@ -0,0 +1,154 @@
+
+1. If `kcl.mod` not exist, search the path `./a/b/c/d` from the current directory.
+2. If the current directory search fails, search from the root path `ROOT_PATH/a/b/c/d`, else raise a import error.
+
+The definition of the root path `ROOT_PATH` is the directory corresponding to the `kcl.mod` file from the current directory.
+
+Code structure:
+
+```
+. 
+└── root
+    ├── kcl.mod
+    ├── model
+    │   ├── model1.k
+    |   ├── model2.k
+    │   └── main.k
+    ├── service
+    │   └── service1.k
+    └── mixin
+        └── mixin1.k
+```
+
+### Builtin Package
+
+KCL has a collection of builtin packages such as `math`, `regex`, etc. To use a builtin package, import it directly and invoke the functions using its qualified identifier. For instance,
+
+```python
+import regex
+
+image = "nginx:1.14.2"
+is_match = regex.match(image, "^[a-zA-Z]+:\d+\.\d+\.\d+$")
+
+```
+
+The output YAML is
+
+```yaml
+image: nginx:1.14.2
+is_match: true
+```
+
+### Plugin Package
+
+KCL also has a collection of plugin packages such as `hello`, `project_context`, etc. To use a plugin package, import it with a `kcl_plugin.` package path prefix and invoke the functions using its qualified identifier. For instance,
+
+```python
+import kcl_plugin.hello
+
+result = hello.add(1, 1)
+```
+
+The output YAML is
+
+```yaml
+result: 2
+```
+
+### Main Package
+
+In KCL, the composition of the main package is usually determined by the compiler parameters. This is because the KCL schema and constraints can be split across files in the package, or even organized across directories, considering the convenience of writing and maintaining the configuration in isolated blocks.
+
+#### Files belonging to a main package
+
+It is up to the user to decide which configurations and constraints to use using the KCL command line. For example,
+
+```bash
+kcl file1.k file2.k
+```
+
+Thus, the main package contains two KCL files named `file1.k` and `file2.k`.
+
+If KCL is told to load the files for a specific directory, for example:
+
+```bash
+kcl ./path/to/package
+```
+
+It will only look KCL files with `.k` suffix and ignore files with `_` prefix or `_test.k` into the main package.
+
+In addition, we can set main package files through configuring the command-line compilation setting file (e.g., `kcl.yaml`) as follows:
+
+```yaml
+kcl_cli_configs:
+  files:
+    - file1.k
+    - file2.k
+```
+
+```bash
+kcl -Y kcl.yaml
+```
+
+> Note: If we do not specify any input files for KCL, KCL will find the default `kcl.yaml` from the command line execution path to read the input file. Besides, if we tell KCL both the input files and the compilation setting file, KCL will take input files entered by the user as the final value.
+
+```bash
+# Whether the 'files' field is configured in `kcl.yaml` or not, the final value of input files is ["file1.k", "file2.k"]
+kcl -Y kcl.yaml file1.k file2.k
+```
+
+## The relationship and difference between `kcl.mod` and `kcl.yaml`
```

Comment:
I don't think it's a good idea to compare these two concepts together in the `concepts` session since the correlation between these two concepts is not so logically significant, and put them together here may lead to confusion.

Maybe it's better to introduce their typical usages respectively here and then distinguish the differences between the two concepts in the `FAQ` session?